server/src/infra/ldap_handler.rs

```diff
@@ -1523,6 +1565,144 @@ mod tests {
+                        vals: vec!["Bôb Böbberson".to_string()],
+                    },
+                    LdapPartialAttribute {
+                        atype: "createtimestamp".to_string(),
+                        vals: vec![chrono::Utc.timestamp(0, 0).to_rfc3339()],
+                    },
+                ],
+            }),
+            // "objectclass", "dn", "uid", "cn", "member", "uniquemember"
+            LdapOp::SearchResultEntry(LdapSearchResultEntry {
+                dn: "cn=group_1,ou=groups,dc=example,dc=com".to_string(),
+                attributes: vec![
+                    LdapPartialAttribute {
+                        atype: "objectclass".to_string(),
+                        vals: vec!["groupOfUniqueNames".to_string()],
+                    },
+                    LdapPartialAttribute {
+                        atype: "dn".to_string(),
+                        vals: vec!["cn=group_1,ou=groups,dc=example,dc=com".to_string()],
+                    },
+                    // UID
+                    LdapPartialAttribute {
+                        atype: "uid".to_string(),
+                        vals: vec!["group_1".to_string()],
+                    },
+                    LdapPartialAttribute {
+                        atype: "cn".to_string(),
+                        vals: vec!["group_1".to_string()],
+                    },
+                    //member / uniquemember : "uid={},ou=people,{}"
+                    LdapPartialAttribute {
+                        atype: "member".to_string(),
+                        vals: vec![
+                            "uid=bob,ou=people,dc=example,dc=com".to_string(),
+                            "uid=john,ou=people,dc=example,dc=com".to_string(),
+                        ],
+                    },
+                    LdapPartialAttribute {
+                        atype: "uniquemember".to_string(),
+                        vals: vec![
+                            "uid=bob,ou=people,dc=example,dc=com".to_string(),
+                            "uid=john,ou=people,dc=example,dc=com".to_string(),
+                        ],
+                    },
+                ],
+            }),
+            make_search_success(),
+        ];
+
+        assert_eq!(ldap_handler.do_search(&request).await, expected_result);
+
+        let request2 =
+            make_search_request("dc=example,dc=com", LdapFilter::And(vec![]), vec!["*", "*"]);
+
+        assert_eq!(ldap_handler.do_search(&request2).await, expected_result);
+
+        let request3 = make_search_request(
+            "dc=example,dc=com",
+            LdapFilter::And(vec![]),
+            vec!["*", "+", "+"],
+        );
+
+        assert_eq!(ldap_handler.do_search(&request3).await, expected_result);
+
+        let request4 =
+            make_search_request("dc=example,dc=com", LdapFilter::And(vec![]), vec![""; 0]);
+
+        assert_eq!(ldap_handler.do_search(&request4).await, expected_result);
```

Comment:
Maybe add a test with `vec!["*", "uid", "distinguishedname"]`

Which brings the question: should we dedupe `dn` vs `distinguishedname`? I think it's okay not to.

Reply:
probably better to not dedup those two, even though they're the same, in the same sense that `uid` and `cn` return the same value.

Although I imagine that the dn <-> distinguishedname alias is probably actually in the RFC somewhere, in the rare case someone requests both, they're likely going to be looking for whatever they ask for by key.